visualize-admin/visualization-tool
# feat: Add a way to retrieve metadata from multiple cubes

This PR adapts the application logic to support retrieving metadata from multiple cubes at once. As we don't yet a design for this feature, there are some `FIXME: adapt to design` comments in places that would need to be revisited once ready.

### GQL / SPARQL
- Removed `DataCubeMetadata` query (replaced by `DataCubesMetadata`)
- Changed the way we fetch cube metadata from `DESCRIBE`ing cube shape to only fetching what we actually need via a dedicated, more performant query. To further improve performance and limit resolver chains, I've extracted the dimension types from GQL to TypeScript. It also means that we now explicitly parse them inside the resolver and don't need to fire additional queries to get theme and creator labels
- Cleaned up the repository from fetching cube metadata in places that didn't require it

---
## Context
This PR is a second part of preparations for merging the cubes. For the technical concept, see [this Notion document](https://www.notion.so/interactivethings/Combine-two-cubes-in-the-same-chart-d5711b414a7747e79edfc3cfe17d2c9b).

- [x] `DataCubesComponents`
- [x] `DataCubesMetadata`
- [ ] `DataCubesObservations`

## How to test
1. Click around in the application and make sure nothing is broken 💥 
2. Open [GQL playground](https://visualization-tool-git-feat-multiple-cubes-metadata-ixt1.vercel.app/api/graphql) and send e.g. the following query to see that metadata from both cubes was fetched.

**Query**
```graphql
query DataCubesMetadata(
  $sourceType: String!
  $sourceUrl: String!
  $locale: String!
  $filters: [DataCubeFilter!]!
) {
  dataCubesMetadata(
    sourceType: $sourceType
    sourceUrl: $sourceUrl
    locale: $locale
    filters: $filters
  )
}
```

**Variables**
```js
{
  "sourceType": "sparql",
  "sourceUrl": "https://lindas.admin.ch/query",
  "locale": "en",
  "filters": [{ "iri": "https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/9", "latest": true }, { "iri": "https://environment.ld.admin.ch/foen/ubd0104/6", "latest": true }]
}
```
